app/profile/components/ProfileOverview.tsx

```diff
@@ -1,86 +1,525 @@
-import { Badge } from "@components/ui/badge";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { ContactType } from "@prisma/client";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@components/ui/form";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "@components/ui/alert-dialog";
+import { Divider } from "@tremor/react";
 import { Button } from "@components/ui/button";
 import { Avatar, AvatarImage, AvatarFallback } from "@components/ui/avatar";
+import { BirthDatePicker } from "./datePicker";
+import { Textarea } from "@components/ui/textarea";
+import { Input } from "@components/ui/input";
+import { ProfileSchema } from "@lib/schemas/profile";
+import { UserSchema } from "@lib/schemas/user";
+import {
+  ContactSchema,
+  ContactSchemaWithProfileId,
+} from "@lib/schemas/contact";
+import { api } from "trpc/react";
+import { toast } from "sonner";
 
 interface ProfileOverviewProps {
   user: Prisma.UserGetPayload<{ include: { profile: true } }>;
+  contacts: Prisma.ContactGetPayload<object>[];
 }
 
-export function ProfileOverview({ user }: ProfileOverviewProps) {
+const ContactInputSchema = z.object({
+  linkedIn: z.string().url().or(z.literal("")),
+  gitHub: z.string().url().or(z.literal("")),
+  phone: z
+    .string()
+    .regex(/^\+?[0-9]+$/)
+    .or(z.literal("")),
+});
+
+export function ProfileOverview({ user, contacts }: ProfileOverviewProps) {
   const profile = user.profile.at(0);
 
-  return (
-    <div className="space-y-8">
-      <div className="mb-12 flex flex-col space-y-6">
-        <div className="flex justify-between">
-          <div className="flex items-center gap-4">
-            <Avatar className="h-28 w-28">
-              <AvatarImage src={user.image ?? undefined} />
-              <AvatarFallback>
-                <User />
-              </AvatarFallback>
-            </Avatar>
-            <div className="flex flex-col gap-3">
-              <h1 className="scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl">
-                {user.name}
-              </h1>
-              <a
-                href={`mailto:${user.email}`}
-                className="text-muted-foreground underline"
-              >
-                {user.email}
-              </a>
-            </div>
-          </div>
+  const linkedInContact = contacts.find(
+    (contact) => contact.type === ContactType.LINKEDIN,
+  );
+  const gitHubContact = contacts.find(
+    (contact) => contact.type === ContactType.GITHUB,
+  );
+  const phoneContact = contacts.find(
+    (contact) => contact.type === ContactType.PHONE,
+  );
 
-          <Button>
-            <PencilSquareIcon className="mr-2 w-5" />
-            Edit
```

Comment:
We need a way for profiles to be created in order to edit them in the first place. The edit profile button seems to not exist if there is no active profile. Would be great if the edit profile also creates the profile if it does not yet exist (for simplicity purposes)

Reply:
So please re-add edit profile